lib/policies/oauth2/oauth2.js

```diff
@@ -172,6 +171,13 @@ module.exports = function () {
 
 function setHeaders (req, headerObject, namespace) {
```

Comment:
How about `filterHeaders` instead of `setHeaders`? That really gives a clear idea of what's happening.